### DISCUSSION TITLE
Component Select - New approach proposed

### DISCUSSION THREAD

**Discussion by turolopezsanabria** [SUI-Components/sui-components#1565] [Ideas and new features]
As discussed many times, and again during our [210609 - Weekly Sync](https://github.com/SUI-Components/sui-components/discussions/1557), I think it's time to tackle many issues related with the component `select`.

**Proposal**

Approach the component select and the many applications around it from scratch.

Instead of linking the select input to an opinionated display (dropdown list), let the select to simply behave as an input that calls different elements depending on the need:

- `Select`: Just the select input
- `Select native`: The input + Will display a `native picker`
- `Select popover`: The input + Will display a `popover` when clicked 
- `Select dropdown`: The input + Will display a `dropdown` list when clicked 
- `Select modal`: The input + Will display a `modal` when clicked 
- `Select drawer`: The input + Will display a `drawer` when clicked 
- etc

**Important stuff to take into consideration**

- Responsive && || Adaptive
- etc

**Comment by ChrisDesignOps** [comment]
Is it possible that the Select component has no problem? 
Could the dropdown components have to accept more "custom options"? or/and the possibility to be full screen on mobile as well?
(Sorry for maybe make more bigger the discussion)

**Comment by ChrisDesignOps** [comment]
Maybe we can create props for customize the DropdownList position (idk if applies to Select&Autosuggest component or on DropdownList directly) but here a visual idea..
<img width="1401" alt="Dropdown Positions" src="https://user-images.githubusercontent.com/81374770/170252289-537cebec-ac0a-4588-829b-ac867ef7a1e9.png">
